models/containerized_model/base_images/Dockerfile.cortex-certifai-python-model-base

```diff
@@ -0,0 +1,95 @@
+# Args:
+# * TOOLKIT_PATH - Path containing Certifai Toolkit 'packages' folder
+# * PY_VERSION - Python version to install. Defaults to 3.8
+# * MINICONDA_URL - URL for installing Miniconda in the container (the binary is platform specific!)
+#                   Defaults to x86_64 linux binary - see https://docs.conda.io/en/latest/miniconda.html
+
+####################################################
+## Build
+####################################################
+FROM registry.access.redhat.com/ubi9/ubi-minimal:latest AS build
+
+USER root
+
+# Library updates & Install wget
+RUN microdnf update -y && \
+    microdnf install wget findutils tar -y && \
+    microdnf clean all -y
+
+# Install miniconda and create an empty Python env
+ARG MINICONDA_URL=https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+RUN wget --quiet $MINICONDA_URL -O /tmp/miniconda.sh && \
+    /bin/bash /tmp/miniconda.sh -b -p /conda -u && \
+    rm /tmp/miniconda.sh
+
+# Add conda to path
+ENV PATH=/conda/bin:$PATH
+
+# Copy Certifai Toolkit packages and other requirements.txt
+ARG TOOLKIT_PATH=certifai_toolkit
+COPY $TOOLKIT_PATH/packages/all/cortex-certifai-common* \
+     $TOOLKIT_PATH/packages/all/cortex-certifai-model-sdk* \
+     requirements.txt /tmp
+
+# Create Python env and install conda-pack
+ARG PY_VERSION=3.8
+RUN conda create -n certifai python=$PY_VERSION -y && \
+    conda install conda-pack
+
+# Install Certifai Packages and other requirements, then use conda-pack to save the env in a standalone environment (/venv).
+RUN conda run -n certifai pip install --no-cache-dir -r /tmp/requirements.txt \
+       $(find /tmp -name cortex-certifai-common-*.zip)[s3,gcp,azure] \
+       $(find /tmp -name cortex-certifai-model*.zip) && \
+    conda-pack -n certifai --ignore-missing-files -o /tmp/env.tar && \
+    mkdir /venv && \
+    cd /venv && tar xf /tmp/env.tar && rm /tmp/env.tar && \
+    /venv/bin/conda-unpack
+
+
+####################################################
+## Runtime
+####################################################
+FROM registry.access.redhat.com/ubi9/ubi-minimal:latest AS runtime
+
+USER root
+
+# Library Updates
+RUN microdnf update -y && \
+    microdnf clean all -y
+
+LABEL release=1 \
+  name="cortex-certifai-model-python$PY_VERSION-base" \
+  vendor="CognitiveScale" \
+  version=1 \
+  summary="Cortex Certifai Prediction Servie Base Image" \
+  description="Cortex Certifai Enterprise Prediction Servie Base Image" \
+  com.cognitivescale.license_terms="https://www.cognitivescale.com/legal-information/"
+
+ # Create .bashrc file and folders for Python/Pip that belong to the non-root user
+RUN touch //.bashrc && \
+    chown -R 1001 //.bashrc && chmod u+rwx //.bashrc && \
```

Comment:
As mentioned in the PR description, the images are _similar_ to the [Certifai base images](https://github.com/CognitiveScale/certifai/blob/develop/docs/packaging.md#docker-containers). Because the images are running as non-root user, they do not have permissions to write to most of the filesystem. Unlike the Certifai apps, the user does have sufficient permissions to run `pip install`

Reply:
The [Tekton Task](https://github.com/CognitiveScale/certifai/blob/4729-spike-tekton/model-packaging/tekton-tasks/c6i-build-prediction-service-task.yaml#L21) does create a few files and move things around, but this is done on the shared volume (which should be a k8s volume mount). So theoretically we should be good with permissions but this needs testing. Is there a example docker image that I can use to test this?

Reply:
Yeah, it would be good to test it out to end-to-end. There's some instructions in the PR description to build it locally

> For local testing, you will need to [download the toolkit from GoCD](https://gocd.dci-dev.dev-eks.insights.ai/go/pipeline/activity/certifai-toolkit) and place it in the artifacts/ folder. Then run:
```
SKIP_CONDA=true ./build.sh local-docker-builder
```